File: docs/guides/php/src/enableFilterPromote.php
<?php

require __DIR__.'/../vendor/autoload.php';

use Algolia\AlgoliaSearch\Model\Search\Anchoring;
use Algolia\AlgoliaSearch\Model\Search\Condition;
use Algolia\AlgoliaSearch\Model\Search\Consequence;
use Algolia\AlgoliaSearch\Model\Search\Rule;

$condition = (new Condition())
    ->setAnchoring((new Anchoring())::IS)
    ->setPattern('{facet:brand}')
;

$consequence = (new Consequence())
    ->setFilterPromotes(true)
;

$rule = (new Rule())
    ->setObjectID('rule_with_filterPromotes')
    ->setConditions([$condition])
    ->setConsequence($consequence);

File: docs/guides/go/src/enableFilterPromote.go
package main

import (
	"fmt"

	"github.com/algolia/algoliasearch-client-go/v4/algolia/next/search"
)

func enableFilterPromote() {
	condition := search.NewCondition().
		SetPattern("{facet:brand}").
		SetAnchoring(search.ANCHORING_IS)

	rule := search.NewRule(
		"rule_with_filterPromotes",
		*search.NewConsequence().SetFilterPromotes(true),
	).SetEnabled(true).SetConditions([]search.Condition{*condition})

	fmt.Printf("Rule: %#v\n", rule)
}
